pandas/core/indexes/range.py

```diff
@@ -443,6 +443,63 @@ def join(self, other, how='left', level=None, return_indexers=False,
         return super(RangeIndex, self).join(other, how, level, return_indexers,
                                             sort)
 
+    def append(self, other):
+        """
+        Append a collection of Index options together
+
+        Parameters
```

Comment:
yes, more or less ready, but it involves many more changes/files and probably some discussion, if it's OK I will open another PR immediately after this